components/common/src/templating/hooks.rs

```diff
@@ -756,11 +756,7 @@ echo "The message is Hola Mundo"
         assert_eq!(hook.compile(&service_group, &ctx).unwrap(), true);
 
         let post_change_content = file_content(&hook);
-        let expected = r#"#!/bin/bash
-
-echo "The message is Hello"
-"#;
-        assert_eq!(post_change_content, expected);
+        assert!(post_change_content.contains("The message is Hello"));
```

Comment:
I think what was happening is that the Supervisor is writing templated files to disk using the newline convention of the platform it's running on. This test (which needs some work to become a "real" unit test, which would help out a lot) is just trying to verify that we're inserting the appropriate data in at our templating markers, which is why we recommended this fix to the test.

@mwrock Are there deeper changes to the Supervisor to more explicitly regulate this behavior? Is there benefit (or worse, danger) in doing this on Windows?

Reply:
Actually I believe the error buildkite was running into was that `CRLF` were injected into the "before" state from disk. When writing lines to files via most APIs and including rust, it only writes `LF`s.

I think this can all go away if we set autocrlf to `false` before we clone. The default on Windows is `true` which will replace `LF`s with `CRLF`s in the checked out code. Appveyor sets it to `input` which replaces `CRLF`s with `LF`s when writing the file to git. For us I think `false` is ideal, it basically just leaves everything alone.

Reply:
@mwrock that was the original tactic I looked into, and set the default up so autocrlf is `false`. However, this had no effect on the actual test, because it appears that the rust library that writes the file to disk is the writing with dos-native file format, with CRLFs. After talking to @christophermaier and @raskchanky about it, we decided this was an ok compromise for the short term until we can rewrite these tests to be more "unit-y". Does that seem reasonable?

Reply:
The rust library will not write `CR`s unless explicitly told to. The test error here is because git is injecting `CR`s into `common/tests/fixtures/hooks/hook_templates/install`.

I have tried this on a fresh AWS VM with a default git config (that has `autocrlf` set to `true`) ant the failute is:

```
---- templating::hooks::tests::compile_a_hook stdout ----
test_service.test_group(HK): Modified hook content in C:\Users\ADMINI~1\AppData\Local\Temp\2\.tmpE5oMiy\install
thread 'templating::hooks::tests::compile_a_hook' panicked at 'assertion failed: `(left == right)`
  left: `"#!/bin/bash\r\n\r\necho \"The message is Hello\"\r\n"`,
 right: `"#!/bin/bash\n\necho \"The message is Hello\"\n"`', components\common\src\templating\hooks.rs:763:9
```
Viewing `common/tests/fixtures/hooks/hook_templates/install` in vscode on the VM I see CRLF line endings and if I change to `LF`, the test passes.

While I don't think changing this individual test to be a terrible thing, I think having an environment where characters are inserted into source that were not commited is. :)

Reply:
Yep I verified that my git config was *not* actually doing the expected thing, working on a fix now!

Reply:
@mwrock I got it - embarrassingly I had forgotten to tell the ASG to use the new AMI :P

Reply:
sweeeeeeeeeeeeeeet!